proxy-socket-core/src/main/java/net/airvantage/proxysocket/core/v2/ProxyProtocolV2Decoder.java

```
@@ -0,0 +1,192 @@
+            }
+        }
+
+        int pos = offset + SIG.length;
+
+        int verCmd = data[pos++] & 0xFF; // version/command
+        int version = (verCmd >> 4) & 0x0F;
+        if (version != 2) throw new ProxyProtocolParseException("Invalid version");
+        int cmd = verCmd & 0x0F;
+        ProxyHeader.Command command = cmd == 0x00 ? ProxyHeader.Command.LOCAL : ProxyHeader.Command.PROXY;
+
+        int famProto = data[pos++] & 0xFF;
+        int fam = famProto & 0xF0;
+        int proto = famProto & 0x0F;
+
+        ProxyHeader.AddressFamily af = parseAddressFamily(fam);
+        ProxyHeader.TransportProtocol tp = parseTransportProtocol(proto);
+
+        int variableLength = ((data[pos++] & 0xFF) << 8) | (data[pos++] & 0xFF);
+
+        int headerLen = SIG.length + 4 + variableLength;
+        if (headerLen > length) throw new ProxyProtocolParseException("Insufficient data for header");
+
+        int addrStart = pos;
+        AddressPair addresses = null;
+
+        if (command == ProxyHeader.Command.PROXY) {
+            addresses = parseAddresses(data, pos, af, tp, variableLength);
+            if (addresses != null) {
+                pos = addresses.newPos;
+            }
+        }
+
+        int consumed = pos - addrStart;
+        int tlvLen = Math.max(0, variableLength - consumed);
+        List<Tlv> tlvs = parseTlvs(data, pos, tlvLen);
+
+        InetSocketAddress src = addresses != null ? addresses.src : null;
+        InetSocketAddress dst = addresses != null ? addresses.dst : null;
+
+        return new ProxyHeader(command, af, tp, src, dst, tlvs, headerLen);
+    }
+
+    private static ProxyHeader.AddressFamily parseAddressFamily(int fam) {
+        return switch (fam) {
+            case 0x10 -> ProxyHeader.AddressFamily.INET4;
+            case 0x20 -> ProxyHeader.AddressFamily.INET6;
+            case 0x30 -> ProxyHeader.AddressFamily.UNIX;
+            default -> ProxyHeader.AddressFamily.UNSPEC;
+        };
```

We should add case 0x00 for UNSPEC
And default should raise an exception